### PR TITLE
gepifanio/c04-iac02

### PR DESCRIPTION
**Pull request by gepifanio**
# Description

Exercise (e.g. C01-E01): c04-iac02

Student Name: gepifanio

> A brief explanation of what has been done.

## Type of change

- [x] Exercise Submission (one exercise per PR)
  - [x] This PR is just for **one** exercise of a class
  - [x] PR name follows the pattern `<github-username>/<exercise-number>`
  - [x] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/04class/exercises/c04-iac02/gepifanio/output.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    value        = aws_subnet.subnet_2_private.cidr_block
+}
+
+output "subnet_3_public" {
+    value        = aws_subnet.subnet_3_public.cidr_block
+}
+
+output "subnet_4_public" {
+    value        = aws_subnet.subnet_4_public.cidr_block
+}
```
Comment (denstorti) [inline classes/04class/exercises/c04-iac02/gepifanio/output.tf:19]: it is possible to group this input in a more complex data structure using `list`, `map` and `object`. So you'd have a more generic and flexible input

File: classes/04class/exercises/c04-iac02/gepifanio/variables.tf [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+variable "az_southeast_a" {
+    type    = string
+    default = "ap-southeast-2a"
+}
+
+variable "az_southeast_b" {
+    type    = string
+    default = "ap-southeast-2b"
+}
+
```
Comment (denstorti) [inline classes/04class/exercises/c04-iac02/gepifanio/variables.tf:15]: tip: it is possible to fetch these values and remove them from input:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/availability_zones

File: classes/04class/exercises/c04-iac02/gepifanio/output.tf (hunk 1 above)
Comment (gepifanio) [reply classes/04class/exercises/c04-iac02/gepifanio/output.tf]: Thanks for the tip !

File: classes/04class/exercises/c04-iac02/gepifanio/variables.tf (hunk 2 above)
Comment (gepifanio) [reply classes/04class/exercises/c04-iac02/gepifanio/variables.tf]: Awesome!!

**Review by denstorti — APPROVED** [review]
Looks good, thanks Gui

**Pull request merged**